src/app/clusters/scenes/SceneTableImpl.h

```diff
@@ -0,0 +1,240 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <app/clusters/scenes/ExtensionFieldSetsImpl.h>
+#include <app/clusters/scenes/SceneTable.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/CommonIterator.h>
+#include <lib/support/PersistentData.h>
+#include <lib/support/Pool.h>
+
+namespace chip {
+namespace scenes {
+
+using clusterId = chip::ClusterId;
+
+/// @brief Default implementation of handler, handle EFS from add scene and view scene commands for any cluster
+///        The implementation of SerializeSave and ApplyScene were omitted and must be implemented in a way that
+///        is compatible with the SerializeAdd output in order to function with the Default Scene Handler.
+///        It is worth noting that this implementation is very memory consuming. In the current worst case,
+///        (Color control cluster), the Extension Field Set's value pair list TLV occupies 99 bytes of memory
+class DefaultSceneHandlerImpl : public scenes::SceneHandler
+{
+public:
+    static constexpr uint8_t kMaxValueSize = 4;
+    static constexpr uint8_t kMaxAvPair    = 15;
+
+    DefaultSceneHandlerImpl() = default;
+    ~DefaultSceneHandlerImpl() override{};
+
+    /// @brief From command AddScene, allows handler to filter through clusters in command to serialize only the supported ones.
+    /// @param endpoint[in] Endpoint ID
+    /// @param extensionFieldSet[in] ExtensionFieldSets provided by the AddScene Command, pre initialized
+    /// @param serialisedBytes[out] Buffer to fill from the ExtensionFieldSet in command
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not supported, CHIP_ERROR value otherwise
+    virtual CHIP_ERROR SerializeAdd(EndpointId endpoint,
+                                    const app::Clusters::Scenes::Structs::ExtensionFieldSet::DecodableType & extensionFieldSet,
+                                    MutableByteSpan & serializedBytes) override
+    {
+        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair;
+        TLV::TLVWriter writer;
+        TLV::TLVType outer;
+        size_t pairTotal  = 0;
+        uint8_t pairCount = 0;
+
+        // Verify size of list
+        extensionFieldSet.attributeValueList.ComputeSize(&pairTotal);
+        VerifyOrReturnError(pairTotal <= kMaxAvPair, CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
Should this compare to ArrayLength(mAVPairs) instead?  Similar for the other size checks.

Reply:
To my knowledge we don't have a library like "array.hpp" in the matter core lib for a basic array class. It would be nice to add one in the future for this kind of functions. In the mean time, I created a span on the array to get the size and remove the kMaxAvPair from the check by doing:
`Span<app::Clusters::Scenes::Structs::AttributeValuePair::Type>(mAVPairs).size(`
I applied the same change to the other checks.

Reply:
`src/lib/support/CodeUtils.h` defines `ArraySize`.  Sorry, I forgot what the thing was called.

Reply:
Using ArraySize now.